### PR TITLE
add PrivBox framework and implement of DLG and GAN

### PR DESCRIPTION
**Pull request by Yanghello**
add PrivBox framework and model inversion attack implement for DLG and GAN

### CODE REVIEW

File: PrivBox/inversion/gan.py [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+            we update single data roundly for each window size iterations
+        return_epoch(int): Return reconstructed data every 'return_epoch' epochs
+    """
+    params = ["learning_rate_real",
+             "learning_rate_fake",
+             "learning_rate_gen"] 
+
+    def __init__(self, gen_net, label_size, target_label, target_fake_label, fake_dataload):
+        """
+        construct DLGInversionAttack
```
Comment (honshj) [inline PrivBox/inversion/gan.py:53]: doc copied from DLG attack, change doc to GAN

File: PrivBox/metrics/acc.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+import abc
+from .metric import Metric
+
+from paddle.nn.functional import mse_loss
+"""
+Accuracy metric modulus, used for evaluation
+"""
+
+
+class AccMetric(Metric):
```
Comment (honshj) [inline PrivBox/metrics/acc.py:25]: rename to Accuracy is OK?

File: PrivBox/metrics/mse.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+import abc
+
+from paddle.nn.functional import mse_loss
+
+from .metric import Metric
+"""
+Mean square error metrics modulus, used for evaluation
+"""
+
+class MSEMetric(Metric):
```
Comment (honshj) [inline PrivBox/metrics/mse.py:25]: Mse is OK?

**Review by honshj — APPROVED** [review]
LGTM

**Review by honshj — APPROVED** [review]
LGTM
